src/components/shared/LoadingButton.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import _Button from '@material-ui/core/Button';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Translate from '../Translate/Translate.react';
+import PropTypes from 'prop-types';
+
+const Button = styled(_Button)`
+  width: 10rem;
+`;
+
+class LoadingButton extends Component {
+  state = {
+    loading: false,
+  };
+
+  static propTypes = {
+    buttonText: PropTypes.string,
+    onConfirm: PropTypes.func,
+    isDisable: PropTypes.bool,
+    color: PropTypes.string,
+    variant: PropTypes.string,
+  };
+
+  handleAction = () => {
+    this.setState(
+      {
+        loading: true,
+      },
+      () => {
+        this.props.onConfirm();
+      },
+    );
+  };
+
+  render() {
+    const { loading } = this.state;
```

Comment:
No need to keep loading in the state. Assume it to be coming in props. And also let the name be `Button`.

Reply:
Should I incorporate simple Button and LoadingButton in one file itself

Reply:
Assume, the default value of loading to be false. Then single component would work.